### PR TITLE
Try to fix alignof under Windows Compiler

### PR DESCRIPTION
**Pull request by gitbuda**
- [x] Extend README because it's also possible to compile mgclient without MinGW
- [x] Add Windows Compiler to CI

Very interesting resources to read:
* [vcpkg OpenSSL linking issue](https://github.com/microsoft/vcpkg/issues/24291)
* [Microcoft Compiler versions explained](https://blog.knatten.org/2022/08/26/microsoft-c-versions-explained/)

### CODE REVIEW

**Review by jbajic — APPROVED** [review]

File: src/mgsession.c [hunk 1]
```diff
@@ -23,6 +22,9 @@
 #ifndef _MSC_VER
 #include <unistd.h>
 #endif
+#if !defined(_WIN32) || !defined(_MSC_VER)
+#include <stdalign.h>
+#endif
```
Comment (antaljanosbenjamin) [inline src/mgsession.c:27]: Maybe unify these two.

File: src/mgallocator.c [hunk 2]
```diff
@@ -15,12 +15,17 @@
...
+// https://docs.microsoft.com/en-us/cpp/cpp/alignment-cpp-declarations?view=msvc-170
+#if !defined(_WIN32) || !defined(_MSC_VER)
 #include <stdalign.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
 #ifdef _MSC_VER
+#define alignof __alignof
```
Comment (antaljanosbenjamin) [inline src/mgallocator.c:28]: I think the conditions here should be the same, as both of them should be there or none of them.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

**Comment by gitbuda** [comment]
@antaljanosbenjamin I think I left that because of MinGW vs MSC, but yea, ifdefs could probably be improved + I've added to update CI config as well. Will try to do that in the following days 😄

File: src/mgsession.c (hunk 1 above)
Comment (gitbuda) [reply src/mgsession.c]: I've polished a bit, I'll add the CI job and merge when everything is compiling

File: src/mgallocator.c (hunk 2 above)
Comment (gitbuda) [reply src/mgallocator.c]: I've polished a bit, I'll add the CI job and merge when everything is compiling

**Review by antaljanosbenjamin — APPROVED** [review]

**Pull request merged**